dockerspawner/dockerspawner.py

```diff
@@ -165,9 +187,12 @@ def _env_default(self):
             JPY_USER=self.user.name,
             JPY_COOKIE_NAME=self.user.server.cookie_name,
             JPY_BASE_URL=self.user.server.base_url,
-            JPY_HUB_PREFIX=self.hub.server.base_url,
-            JPY_HUB_API_URL=self.hub.api_url,
+            JPY_HUB_PREFIX=self.hub.server.base_url
         ))
+        if self.hub_ip_connect:
+           env.update(dict(JPY_HUB_API_URL=self._public_hub_api_url()))
```

Comment:
You don't need to use `update` for setting just one key. You can use `env['JPY_HUB_API_URL'] = url`